BitwardenShared/Core/Vault/Services/Fido2CredentialStoreServiceTests.swift

```diff
@@ -0,0 +1,309 @@
+    // MARK: Properties
+
+    var cipherService: MockCipherService!
+    var clientService: MockClientService!
+    var subject: Fido2CredentialStoreService!
+
+    // MARK: Setup & Teardown
+
+    override func setUp() {
+        super.setUp()
+
+        cipherService = MockCipherService()
+        clientService = MockClientService()
+
+        subject = Fido2CredentialStoreService(
+            cipherService: cipherService,
+            clientService: clientService
+        )
+    }
+
+    override func tearDown() {
+        super.tearDown()
+
+        cipherService = nil
+        clientService = nil
+        subject = nil
+    }
+
+    // MARK: Tests
+
+    /// `.allCredentials()` returns all credentials decrypted which are active
+    /// and have Fido2 credentials in it.
+    func test_allCredentials() async throws {
+        cipherService.fetchAllCiphersResult = .success([
+            .fixture(deletedDate: Date.distantPast, id: "1"),
+            .fixture(id: "2", type: .card),
+            .fixture(id: "3", type: .identity),
+            .fixture(id: "4", type: .login),
+            .fixture(
+                id: "5",
+                login: .fixture(
+                    fido2Credentials: [
+                        .fixture(),
+                    ]
+                ),
+                type: .login
+            ),
+            .fixture(id: "6", type: .secureNote),
+        ])
+
+        let result = try await subject.allCredentials()
+
+        XCTAssertTrue(result.count == 1)
+        XCTAssertTrue(result[0].id == "5")
+    }
+
+    /// `.allCredentials()` throws when fetching ciphers.
+    func test_allCredentials_throwsFetchingCiphers() async throws {
+        cipherService.fetchAllCiphersResult = .failure(BitwardenTestError.example)
+
+        await assertAsyncThrows(error: BitwardenTestError.example) {
+            _ = try await subject.allCredentials()
+        }
+    }
+
+    /// `.allCredentials()` throws when decrypting ciphers.
+    func test_allCredentials_throwsDecryptingCiphers() async throws {
+        cipherService.fetchAllCiphersResult = .success([
+            .fixture(
+                id: "1",
+                login: .fixture(
+                    fido2Credentials: [
+                        .fixture(),
+                    ]
+                ),
+                type: .login
+            ),
+        ])
+        clientService.mockVault.clientCiphers.decryptResult = { _ in
+            throw BitwardenTestError.example
+        }
+
+        await assertAsyncThrows(error: BitwardenTestError.example) {
+            _ = try await subject.allCredentials()
+        }
+    }
+
+    /// `.findCredentials(ids:ripId:)` returns the login ciphers that are active, have Fido2 credentials
+    /// and match the `ripId` and the credential `ids` if any.
+    func test_findCredentials() async throws {
+        let expectedRpId = Fido2CredentialAutofillView.defaultRpId
+        let expectedCredentialId = Data((0 ..< 16).map { _ in 1 })
```

Comment:
🎨 There's a `Data` initializer that should give you this same behavior. 
```suggestion
        let expectedCredentialId = Data(repeating: 1, count: 16)
```